### PR TITLE
Fixing test reliablity for DownloadArtifactsTfsGit

### PR DESCRIPTION
**Pull request by SunnyKumarR**

### CODE REVIEW

File: Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ var branch = tl.getInput("branch");
 var commitId = tl.getInput("version");
 var downloadPath = tl.getInput("downloadPath");
 var tfsEndpoint = getEndpointDetails("connection");
+var retryLimit = 4;
```
Comment (omeshp) [inline Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js]: use VSTS_HTTP_RETRY variable

File: Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js [hunk 2]
```diff
@@ -24,6 +25,25 @@ if(error){
     tl.exit(1);
 }
 
+function executeWithRetries (operationName, operation, currentRetryCount) {
+    var deferred = Q.defer()   
+    operation().then((result) => {
+      deferred.resolve(result)
+    }).fail((error) => {
+      if (currentRetryCount <= 0) {
+        tl.error('OperationFailed: git clone')
```
Comment (omeshp) [inline Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js]: use operationName field.

**Review by omeshp — APPROVED** [review]

File: Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js (hunk 1 above)
Comment (SunnyKumarR) [reply Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js]: Resolved

File: Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js (hunk 2 above)
Comment (SunnyKumarR) [reply Extensions/ExternalTfs/Src/Tasks/DownloadArtifactsTfsGit/downloadTfGit.js]: Resolved

**Pull request merged**